db/change_cache.go

```diff
@@ -59,6 +59,7 @@ type changeCache struct {
 	lock               sync.RWMutex            // Coordinates access to struct fields
 	options            CacheOptions            // Cache config
 	terminator         chan bool               // Signal termination of background goroutines
+	terminated         []TaskStat              // Notifies on each background task termination.
```

Comment:
'terminated' sounds like it should be a boolean value.  Can you switch this to a name that better describes the property?